### PR TITLE
Massive Update

### PR DESCRIPTION
**Pull request by lowercasebtw**
Basically redid everything, and added tons of new converters/fixes.

### CODE REVIEW

File: bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/BedrockPackConverter.java [hunk 1]
```diff
@@ -18,58 +23,57 @@
...
-        PackConverter.out = out;
-        this.INPUT_DIR = input;
+	public BedrockPackConverter(Path input, boolean minify, boolean debug, PrintStream out) {
+		GsonBuilder gsonBuilder = new GsonBuilder().disableHtmlEscaping();
+		if (!minify)
+			gsonBuilder.setPrettyPrinting();
 
-        log("Java");
-        log("Bedrock");
+		gsonBuilder.disableHtmlEscaping();
```
Comment (CoryBorek) [inline bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/BedrockPackConverter.java]: Duplicate line

File: bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/Util.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+public final class Util {
```
Comment (CoryBorek) [inline bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/Util.java]: Is this file in a dependency jar?

File: bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/impl/PackMetaConverter.java [hunk 3]
```diff
@@ -1,76 +1,77 @@
...
+			JsonArray version = new JsonArray();
+			version.add(1);
+			version.add(0);
+			version.add(0);
+			header.add("version", version);
+
+			// min_engine_version
+			JsonArray minVersion = new JsonArray();
+			minVersion.add(1);
+			minVersion.add(16);
```
Comment (CoryBorek) [inline bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/impl/PackMetaConverter.java]: We need to make sure that this is the actual min supported version

File: bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/impl/PurgerConverter.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        super(packConverter);
+    }
+
+    @Override
+    public void convert(Pack pack) throws IOException {
+        File[] files = pack.getWorkingPath().toFile().listFiles();
+        for (File file : files) {
+            Logger.log(file.getPath());
+        }
+    }
```
Comment (CoryBorek) [inline bedrock-conversion/src/main/java/com/agentdid127/resourcepack/bedrock/impl/PurgerConverter.java]: Is there an actual use to this? Just printing doesn't make a ton of sense.

**Review by CoryBorek — CHANGES_REQUESTED** [review]

**Review by lowercasebtw — COMMENTED** [review]
fixed

**Review by CoryBorek — APPROVED** [review]

**Pull request merged**